packages/web/src/store/application/ui/buy-audio/sagas.ts

```diff
@@ -233,25 +236,51 @@ function* getTransactionFees({
       }
     )
     const connection = yield* call(getSolanaConnection)
+
+    // Calculate fees for transfer transaction (legacy transaction)
     const latestBlockhashResult = yield* call(
       [connection, connection.getLatestBlockhash],
       'finalized'
     )
-    for (const transaction of [swapTransaction, transferTransaction]) {
-      if (transaction) {
-        if (!transaction.recentBlockhash) {
-          transaction.recentBlockhash = latestBlockhashResult.blockhash
-        }
-        if (!transaction.feePayer) {
-          transaction.feePayer = rootAccount
-        }
-        const fee = yield* call(
-          [transaction, transaction.getEstimatedFee],
-          connection
-        )
-        transactionFees += fee ?? 5000 // For some reason, swap transactions don't have fee estimates??
+    if (!transferTransaction.recentBlockhash) {
+      transferTransaction.recentBlockhash = latestBlockhashResult.blockhash
+    }
+    if (!transferTransaction.feePayer) {
+      transferTransaction.feePayer = rootAccount
+    }
+    transactionFees +=
+      (yield* call(
+        [transferTransaction, transferTransaction.getEstimatedFee],
+        connection
+      )) ?? 5000
+
+    // Calculate fees for swap transaction (v0 transaction)
+    const lookupTableAccounts: AddressLookupTableAccount[] = []
+    // Need to use for loop instead of forEach to properly await async calls
+    for (let i = 0; i < lookupTableAddresses.length; i++) {
+      const address = lookupTableAddresses[i]
+      if (address === undefined) continue
+      const lookupTableAccount = yield* call(
+        [connection, connection.getAddressLookupTable],
+        new PublicKey(address)
+      )
+      if (lookupTableAccount.value !== null) {
+        lookupTableAccounts.push(lookupTableAccount.value)
+      } else {
+        // Abort if a lookup table account is missing because the resulting transaction
+        // might be too large
+        throw new Error(`Missing lookup table account for ${address}`)
       }
     }
+    const message = new TransactionMessage({
```

Comment:
kinda sad we have to do this twice. If only the transaction handler handled.. well, transactions!